samples/crypto/tls/src/non-secure/tls_credentials_client.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(tls_credentials);
+
+#include <nrf.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <net/socket.h>
+#include <net/net_core.h>
+#include <net/tls_credentials.h>
+#include <linker/sections.h>
+
+#include "certificate.h"
+#include "psa_tls_credentials.h"
+
+#include <tfm_ns_interface.h>
+#include <psa/storage_common.h>
+#include "psa/protected_storage.h"
+
+static unsigned char ca_cert_buf[sizeof(ca_certificate)];
+
+
+/** @brief Function for registering certificates and PSKs in Protected Storage.
+ */
+void tls_store_credentials_in_ps(void)
+{
+	psa_status_t status;
+
+	LOG_INF("Will store Client CA certificate in Protected Storage");
+
+	status = psa_ps_set(PSA_PS_CA_CERTIFICATE_UID,
+			    sizeof(ca_certificate),
+			    ca_certificate,
+			    PSA_STORAGE_FLAG_WRITE_ONCE);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("Failed to store CA certificate to"
+			" Protected Storage. Status: %d", status);
+	}
+}
+
+void tls_set_credentials_from_ps(void)
+{
+	size_t cred_len;
+
+	LOG_INF("Will register Client CA certificate from Protected Storage");
+
+	psa_status_t status = psa_ps_get(PSA_PS_CA_CERTIFICATE_UID,
+			    (size_t) 0,
+			    sizeof(ca_certificate),
+			    ca_cert_buf,
+			    &cred_len);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("Failed to retrieve CA certificate from"
+			" Protected Storage. Status: %d", status);
```

Comment:
If that fails will it stop the execution? Is it indented to continue with the tsl_credential_add?

Reply:
I've updated the sample to terminate if setup of credentials fails